lib/ui/widgets/ata_map.dart

```diff
@@ -0,0 +1,181 @@
+      this.authRange = 100,
+      this.titleMarkedPosition = "Marked Position"});
+
+  @override
+  State<ATAMap> createState() => ATAMapState();
+}
+
+class ATAMapState extends State<ATAMap> {
+  double currentMarkedLat;
+  double currentMarkedLng;
+  double currentLocationLat;
+  double currentLocationLng;
+  bool isMapReady = false;
+  static const int TIMEOUT_PIN_MARKER_MAP_LOADED = 2;
+  static const double DEFAULT_ZOOM = 17;
+  CameraPosition get defaultCamera {
+    return CameraPosition(
+      target: LatLng(widget.centerMapLat, widget.centerMapLng),
+      zoom: DEFAULT_ZOOM,
+    );
+  }
+
+  Completer<GoogleMapController> _controller = Completer();
+  Set<Marker> _markers = Set();
+  Set<Circle> _circles = Set();
+  BitmapDescriptor markedLocationIcon;
+  @override
+  void initState() {
+    super.initState();
+    currentMarkedLat = widget.markedLat;
+    currentMarkedLng = widget.markedLng;
+    _setCustomMapIcons();
+  }
+
+  void _setCustomMapIcons() async {
+    markedLocationIcon = await BitmapDescriptor.fromAssetImage(ImageConfiguration(size: Size(128, 128)), 'assets/images/marked-location-icon.png');
+  }
+
+  Future<Position> getCurrentLocation() async => await Geolocator().getCurrentPosition();
+
+  void _goToCurrentLocation() async {
+    try {
+      Position currentPosition = await getCurrentLocation();
+      final GoogleMapController controller = await _controller.future;
+      controller.animateCamera(
+          CameraUpdate.newCameraPosition(CameraPosition(target: LatLng(currentPosition.latitude, currentPosition.longitude), zoom: DEFAULT_ZOOM)));
+      currentLocationLat = currentPosition.latitude;
+      currentLocationLng = currentPosition.longitude;
+    } on PlatformException catch (e) {
+      if (e.code == 'PERMISSION_DENIED') {
+        showDialog(
+            context: context,
+            builder: (BuildContext context) {
+              return AlertDialog(title: Text("PERMISSION_DENIED"), content: Text('GPS device is OFF, please turn ON !'));
+            });
+      }
+    }
+    setState(() {
+      if (_markers.length == 0) _addMarker(LatLng(currentMarkedLat, currentMarkedLng));
+    });
+  }
+
+  void _addMarker(LatLng point) {
+    setState(() {
+      _markers.clear();
+      _circles.clear();
+      currentMarkedLat = point.latitude;
+      currentMarkedLng = point.longitude;
+      _markers.add(Marker(
+          markerId: MarkerId(point.toString()),
+          position: point,
+          infoWindow: InfoWindow(
+              title: widget.isMoveableMarker ? '$currentMarkedLat, $currentMarkedLng' : '${widget.titleMarkedPosition}',
+              snippet: 'Distance :${_calcDistance()} m -> isCheckinable:${isCheckinable()}'),
+          icon: markedLocationIcon,
+          draggable: widget.isMoveableMarker,
+          onDragEnd: ((newPoint) {
+            if (widget.isMoveableMarker) {
+              _addMarker(newPoint);
+              currentMarkedLat = newPoint.latitude;

```

Comment:
currentMarkedLat = newPoint.latitude;  -> no need to set again, coz inside _addMarker() already has set it at the beginning